dmoj/utils/helper_files.py

```diff
@@ -13,7 +13,7 @@ def mktemp(data):
     return tmp
 
 
-def compile_with_auxiliary_files(filenames, lang=None, compiler_time_limit=None):
+def compile_with_auxiliary_files(filenames, flags=[], lang=None, compiler_time_limit=None, should_cached=True):
```

Comment:
"should_cached" sounds very weird

Reply:
It's also the wrong name as per https://github.com/DMOJ/judge-server/pull/609/files#diff-57b85c3e8735c00d8b1353b8903360ffR281

Reply:
Please test this code